src/orderbook/state.ts

```diff
@@ -28,7 +83,139 @@ export class AccountState {
       )
     }
     this.batchId = batchId
-    // TODO(nlordell): Actually apply events to state.
+
+    for (const ev of events) {
+      const data = ev.returnValues
+      switch (ev.event) {
+        case 'OrderCancellation':
+          this.order(data.owner, parseInt(data.id)).validUntil = await this.batches.forEvent(ev)
```

Comment:
Should be the event's batch id minus 1 https://github.com/gnosis/dex-contracts/blob/master/contracts/BatchExchange.sol#L258 .

Reply:
Good catch!